### PR TITLE
CGMAbsSys.igm_sys updates

### PR DESCRIPTION
**Pull request by jnburchett**
Addresses issues related to updating component data within a given CGM system's igm_sys where updates to various CGM systems associated with the same IGMSystem would propagate to all associated to CGM systems.

### CODE REVIEW

File: pyigm/cgm/utils.py
Comment (profxj) [inline pyigm/cgm/utils.py]: have been advised to avoid copy statements when we can.
Can we?

**Review by profxj — CHANGES_REQUESTED** [review]
See my pretty major comment.

**Comment by profxj** [comment]
As a possible example, see the copy() method
of XSpectrum1D in linetools:

https://github.com/linetools/linetools/blob/master/linetools/spectra/xspectrum1d.py

File: pyigm/cgm/utils.py
Comment (jnburchett) [reply pyigm/cgm/utils.py]: Hmm...part of the issue was that updates were propagating through the AbsComponent objects too!  In fact, merely copying the IGMSystem didn't fix the problem, so the deepcopy was the best solution I could think of.  We could of course reinstantiate AbsComponents, but that's less than ideal for obvious reasons...suggestions?

Comment (profxj) [reply pyigm/cgm/utils.py]: I probably need to better understand the problem here.

If one does update the IGMSystem, wouldn't you want that
to propagate?  If not, I do feel you should re-instantiate.

Comment (jnburchett) [reply pyigm/cgm/utils.py]: Just to clarify, yes, one would want the main attributes of a component (column density, b-value, transitions) to propagate but the same IGM system will have different 'systemic' component velocities when associated to one galaxy (CGM systems) versus another.  The .copy methods are very useful anyway, so I'll go that direction.

**Comment by jnburchett** [comment]
Okay, @profxj, think we should be good to go here.  Note that the changes depend on new linetools PR #445:  https://github.com/linetools/linetools/pull/445

**Comment by profxj** [comment]
This all looks fine.  Merge when you wish.

**Comment by jnburchett** [comment]
Great...will merge once the linetools dependency is clear.

**Comment by profxj** [comment]
If we are clear @jnburchett , please merge.

**Comment by jnburchett** [comment]
Weird issue occurring, and I wonder if it's an astropy bug.  Basically, SkyCoord objects with multiple coordinates fail when transforming, i.e., coords.icrs returns this:

*** TypeError: For this input type lists must contain either int or Ellipsis

Could you try this on your machine?  I was using astropy 2.0.2 and tried upgrading to 2.0.3 but get the same error.  This did not happen until we merged add_richter, but I don't think that's the issue.

**Pull request merged**
